docs/console/incremental-guide.md

```diff
@@ -0,0 +1,586 @@
+
+Create a new file and give it a unique namespace:
+
+```csharp
+using System;
+
+namespace NumberFun
+{
+    // code can go here
+} 
+```
+
+Next, include it in your `Program.cs` file:
+
+```csharp
+using System;
+using NumberFun;
+```
+
+And finally, you can compile it:
+
+`$ dotnet compile`
+
+Now the fun part: making the new file do something!
+
+### Example: A Fibonacci Sequence Generator
+
+Let's say you want to build off of the previous fibonacci example by caching some fibonacci values and add some corecursive flair.  Your code might look something like this:
+
+```csharp
+using System;
+using System.Collections.Generic;
+
+namespace NumberFun
+{
+	public class FibonacciGenerator
+	{
+		private Dictionary<int, int> _cache = new Dictionary<int, int>();
+		
+		private int Fib(int n) => n < 2 ? n : FibValue(n - 1) + FibValue(n - 2);
+		
+		private int FibValue(int n)
+		{
+			if (!_cache.ContainsKey(n))
+			{
+				_cache.Add(n, Fib(n));
+			}
+			
+			return _cache[n];
+		}
+		
+		public IEnumerable<int> Generate(int n)
+		{
+			for (int i = 0; i < n; i++)
+			{
+				yield return FibValue(i);
+			}
+		}
+	}
+}
+```
+
+Note that the use of `Dictionary<int, int>` and `IEnumerable<int>` means incorporating the `System.Collections` library in your `project.json`:
+
+```javascript
+{
+    "version": "1.0.0-*",
+    "compilationOptions": {
+        "emitEntryPoint": true
+    },
+
+    "dependencies": {
+        "Microsoft.NETCore.Runtime": "1.0.1-beta-*",
+        "System.IO": "4.0.11-beta-*",
+        "System.Console": "4.0.0-beta-*",
+        "System.Runtime": "4.0.21-beta-*",
+        "System.Collections": "4.0.11-beta-23516"
+    },
+
+    "frameworks": {
+        "dnxcore50": { }
+    }
+}
+```
+
+Now adjust the `Main()` method in you `Program.cs`:
+
+```csharp
+public static void Main(string[] args)
+{
+    var generator = new FibonacciGenerator();
+    foreach (var digit in generator.Generate(15))
+    {
+        Console.WriteLine(digit);
+    }
+}
+```
+
+Finally, run it!
+
+```
+$ dotnet run
+0
+1
+1
+2
+3
+5
+8
+13
+21
+34
+55
+89
+144
+233
+377
+```
+
+And that's it!
+
+## Using folders to organize code
+
+Say you wanted to introduce some new types to do work on.  You can do this by adding more files and making sure to give them namespaces you can include in your `Program.cs` file.
+
+```
+/MyProject
+|__Program.cs
+|__Type1.cs
+|__Type2.cs
+|__project.json
+```
+
+This works great when the size of your project is relatively small.  However, if you have a larger app with many different data types and potentially multiple layers, you may wish to organize things logically.  This is where folders come into play.  You can either follow along with [the sample project](https://github.com/dotnet/core-docs/samples/core-projects/console-apps/NewTypes) that this guide covers, or create your own files and folders.
```

Comment:
The "NewTypes" name is still concerning for me. I don't think anyone would ever choose a name like that.